operate/schema/src/main/java/io/camunda/operate/schema/IndexSchemaValidator.java

```diff
@@ -135,4 +147,92 @@ public boolean schemaExists() {
       return false;
     }
   }
+
+  /**
+   * Validates existing indices mappings against schema files defined in codebase.
+   *
+   * @return newFields map with the new field definitions per index
+   * @throws OperateRuntimeException in case some fields would need to be deleted or have different
+   *     settings
+   */
+  public Map<IndexDescriptor, Set<IndexMappingProperty>> validateIndexMappings() {
```

Comment:
You could extract methods for the big building blocks of this method to make it better readable